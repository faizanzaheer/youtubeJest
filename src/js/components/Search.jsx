import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

export default class Search extends React.Component {
  constructor() {
    super();
    this.state = {
      value: ""
    }

    this.localHandler = this.localHandler.bind(this)
    // this.localHandler = _.debounce(this.localHandler, 500)
  }

  localHandler(e) {
    const { getVideos } = this.props;
    // run the api with the input value
    this.setState({value: e.target.value})
    getVideos(e.target.value);

  }

  render() {
    return (
      <div className="search col-md-6 col-md-offset-4 form-inline"> 
        <input 
          type="text" 
          className="form-control"
          // ref={(elem)=>this.input=elem}
          onChange={this.localHandler}
          value = {this.state.value}
        />
        <button className="btn">
          <span className="glyphicon glyphicon-search "></span>
        </button>
      </div>
    )
  }
};

Search.propTypes = {
  getVideos: PropTypes.func.isRequired,
}








