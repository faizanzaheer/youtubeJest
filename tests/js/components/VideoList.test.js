import React from 'react';
import  {VideoList}  from '../../../src/js/components/VideoList';


import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme'
Enzyme.configure({ adapter: new Adapter() });

import renderer from 'react-test-renderer';
//this package is provided by jest(replacing react-test)
describe('Testing VideoList....', ()=>{
	let videoList = [{id:{videoId: 1}},{id: {videoId:2}}];
	const selectVideo = jest.fn();
	let wrapper = shallow(
		<VideoList
			selectVideo = {selectVideo}
			videoList = {videoList}
			isLoading= {false}
			error =	{null}	
		/>)
	expect(wrapper).toMatchSnapshot();
	

it('renders the loading message',()=>{
	let wrapper = shallow(
		<VideoList
			selectVideo={selectVideo}
			videoList={videoList}
			isLoading={false}
			error={{message: 'Error while loading Videos'}}
			/>)
	expect(wrapper).toMatchSnapshot();
	});

});

	// it('it runs the getVideos function with correct args', ()=>{
	// 	wrapper.find('input').simulate('change',{target:{value:'java'}})
	// 	expect(getVideos.mock.calls[0][0]).toEqual('java');
	// expect(wrapper).toMatchSnapshot();
	
	// });
