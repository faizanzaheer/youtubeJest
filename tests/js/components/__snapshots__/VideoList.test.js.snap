// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoList
    error={null}
    isLoading={false}
    selectVideo={[MockFunction]}
    videoList={
      Array [
        Object {
          "id": Object {
            "videoId": 1,
          },
        },
        Object {
          "id": Object {
            "videoId": 2,
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <VideoListEntry
          handleSelectedVideo={[MockFunction]}
          video={
            Object {
              "id": Object {
                "videoId": 1,
              },
            }
          }
        />,
        <VideoListEntry
          handleSelectedVideo={[MockFunction]}
          video={
            Object {
              "id": Object {
                "videoId": 2,
              },
            }
          }
        />,
      ],
      "className": "videoList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "handleSelectedVideo": [MockFunction],
          "video": Object {
            "id": Object {
              "videoId": 1,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "handleSelectedVideo": [MockFunction],
          "video": Object {
            "id": Object {
              "videoId": 2,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <VideoListEntry
            handleSelectedVideo={[MockFunction]}
            video={
              Object {
                "id": Object {
                  "videoId": 1,
                },
              }
            }
          />,
          <VideoListEntry
            handleSelectedVideo={[MockFunction]}
            video={
              Object {
                "id": Object {
                  "videoId": 2,
                },
              }
            }
          />,
        ],
        "className": "videoList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "handleSelectedVideo": [MockFunction],
            "video": Object {
              "id": Object {
                "videoId": 1,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "handleSelectedVideo": [MockFunction],
            "video": Object {
              "id": Object {
                "videoId": 2,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Testing VideoList.... renders the loading message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VideoList
    error={
      Object {
        "message": "Error while loading Videos",
      }
    }
    isLoading={false}
    selectVideo={[MockFunction]}
    videoList={
      Array [
        Object {
          "id": Object {
            "videoId": 1,
          },
        },
        Object {
          "id": Object {
            "videoId": 2,
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Error while loading Videos
      </div>,
      "className": "videoList",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Error while loading Videos",
      },
      "ref": null,
      "rendered": "Error while loading Videos",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Error while loading Videos
        </div>,
        "className": "videoList",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Error while loading Videos",
        },
        "ref": null,
        "rendered": "Error while loading Videos",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
